### ISSUE TITLE
LSTMCell with dynamic batch size

### ISSUE THREAD

**Issue by Grigor355** [openvinotoolkit/openvino#11730]
I have a CNN+BiLSTM+LSTMCell model in ONNX. I convert the model to OpenVino IR using the following command 
```
mo --input_model "model.onnx" --input_shape "[-1, 1, h, w]" --output_dir "ov"  --data_type FP32 
```
The IR is generated but when I compile and run prediction, the prediction "hangs" and do not produce anything. When I change the dynamic batch size to some number, the problem disappears. I turned off each part of (CNN, BiLSTM, LSTMCell) and the problem exists only when I add LSTMCell to the model. I tried the benchmark app, the results above were reproduced.

ONNX opset - 15
OS - Ubuntu 18.04
OpenVino Version - 2022.1.0 (installed through pip)
Using Python

**Comment by Iffa-Meah** [comment]
@Grigor355 could you share your model files for us to test on our end?

**Comment by Grigor355** [comment]
Yes, as I mentioned, there is no problem when using static input shape, but my application requires dynamic shape.

**Comment by Grigor355** [comment]
@Iffa-Meah I sent the ONNX model via email. Thanks

**Comment by Iffa-Meah** [comment]
@Grigor355 I had requested access to your files

![image](https://user-images.githubusercontent.com/79643476/170161011-cc38b358-8a6d-416e-bcc4-efaa91548545.png)

**Comment by Iffa-Meah** [comment]
@Grigor355 
I had checked your model.

Static IR -- success inference
<img width="627" alt="11730IRstatic" src="https://user-images.githubusercontent.com/79643476/170407530-d06570cd-4627-4341-93a0-444bf43d6468.png">
<img width="619" alt="11730IRstaticbenchmark" src="https://user-images.githubusercontent.com/79643476/170407563-019d4a90-ad0b-4588-92a8-7e0cf948942f.png">


Dynamic IR -- failed inference

<img width="629" alt="11730IRdynamic" src="https://user-images.githubusercontent.com/79643476/170407624-ad03e898-87b2-4dc7-92fa-3e5e034abe0f.png">
<img width="623" alt="11730irdynamicbenchmark" src="https://user-images.githubusercontent.com/79643476/170407644-585fe306-db29-4460-a9fa-7924873490ea.png">


We will further investigate this and get back to you asap.

**Comment by Grigor355** [comment]
Can you call mo with batch size as dynamic? and then for benchmark_app you can provide -data_shape to avoid this error

**Comment by Grigor355** [comment]
Hi, any news on this part?

**Comment by jgespino** [comment]
@Grigor355 Apologies for the delay, could you try to specify the input shape when running benchmark_app? 

I converted your onnx model using the following command:
`mo --input_model model_g.onnx --input_shape [-1,1,32,100] --data_type FP32`

I was able to run the benchmark_app with different batch sizes as follows:
```
benchmark_app -m model_g.xml -shape input[1,1,32,100]
benchmark_app -m model_g.xml -shape input[2,1,32,100]
benchmark_app -m model_g.xml -shape input[3,1,32,100]
```

**Comment by jgespino** [comment]
@Grigor355 Currently dynamic batching does not support models that contain LSTM operations. As you initially observed, the model works when LSTM op is not present. I've confirmed this by using another model that does not contain LSTM op and converted with `mo --input_model <model> --input_shape "[-1, 1, h, w]"`, using your code above, I was able to run inference and updating the batch size without any issues.

**Comment by Grigor355** [comment]
My model has CNN->BiLSTM -> LSTMcell architecture and the problem appears only when LSTMcell is present, so for CNN->BiLSTM model it works. If OpenVino does not support LSTMCell how does the benchmark app work? Both you and me could run the model with LSTMcell and dynamic batch with benchmark_app -m model_g.xml -shape input[2,1,32,100]
??
